### PR TITLE
[ASUSGAB-1437] Add to DoneXBlock override functionality for staff debug info

### PR DESCRIPTION
**Pull request by vladhrech**
**Description:**
* Inherit DoneXblock from ScorableXBlockMixin
* Add flag to display block under specific learner
* Add functionality to handle overrides and rescore

**YouTrack:** https://youtrack.raccoongang.com/issue/ASUSGAB-1437

- [x] Demo is shown to QA

**Notes:**
After performing merge of this issue ping me to create tag with project-version of the xblock

### CODE REVIEW

**Comment by arsentur** [comment]
We can't merge it to the master branch, because this is customization. Pelase, open new branch from master and change PR parameters to new branch.

File: setup.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ def package_data(pkg, roots):
 
 setup(
     name='done-xblock',
-    version='0.1.0-rg',
+    version='0.1.2-rg',
```
Comment (arsentur) [inline setup.py:20]: Why we choose from 0 to 2?

File: done/done.py [hunk 2]
```diff
@@ -183,3 +201,98 @@ def max_score(self):
         """The maximum raw score of our problem.
         """
         return 1
+
+    # ScorableXBlockMixin methods
+    def set_score(self, score):
```
Comment (arsentur) [inline done/done.py:206]: If this method is identical to the method of parent class, it can be skipped. Any objections?

File: done/done.py [hunk 3]
```diff
@@ -183,3 +201,98 @@ def max_score(self):
...
+        """
+        Sets the score on this block.
+        Takes a Score namedtuple containing a raw
+        score and possible max (for this block, we expect that this will
+        always be 1).
+        """
+        self.score = score
+
+    def get_score(self):
```
Comment (arsentur) [inline done/done.py:216]: ditto

File: done/done.py [hunk 4]
```diff
@@ -183,3 +201,98 @@ def max_score(self):
         """The maximum raw score of our problem.
         """
         return 1
+
+    # ScorableXBlockMixin methods
+    def set_score(self, score):
+        """
+        Sets the score on this block.
```
Comment (arsentur) [inline done/done.py:208]: Title of the docstring should be separated from the body of the docstring by one empty line

File: done/done.py [hunk 5]
```diff
@@ -183,3 +201,98 @@ def max_score(self):
...
+    def get_score(self):
+        """
+        Returns the score currently set on the block.
+        """
+        return self.score
+
+    def weighted_grade(self):
+        """
+        Returns the block's current saved grade multiplied by the block's
+        weight- the number of points earned by the learner.
```
Comment (arsentur) [inline done/done.py:226]: ditto

File: done/done.py [hunk 6]
```diff
@@ -183,3 +201,98 @@ def max_score(self):
...
+            self.done = True
+        else:
+            self.done = False
+        self.runtime.publish(self, "edx.done.toggled", {'done': self.done})
+
+        return {'grade': self.score.raw_earned, 'max_grade': self.score.raw_possible}
+
+    def rescore(self, only_if_higher):
+        """
+        Calculate a new raw score and save it to the block.  If only_if_higher
```
Comment (arsentur) [inline done/done.py]: ditto

**Review by arsentur — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (vladhrech) [reply setup.py:20]: it was missed on last provided version, so current on sga is 0.1.1 according to tag

File: done/done.py (hunk 2 above)
Comment (vladhrech) [reply done/done.py:206]: Default methods are not implemented

File: done/done.py (hunk 3 above)
Comment (vladhrech) [reply done/done.py:216]: Default methods are not implemented

File: setup.py (hunk 1 above)
Comment (arsentur) [reply setup.py:20]: resolved

File: done/done.py (hunk 2 above)
Comment (arsentur) [reply done/done.py:206]: Resolved

File: done/done.py (hunk 3 above)
Comment (arsentur) [reply done/done.py:216]: Done

**Review by arsentur — APPROVED** [review]

**Pull request merged**